### PR TITLE
Feature/google optimize

### PR DESCRIPTION
**Pull request by williamchong007**
@nwingt Please comment on both the implementation of `register-icon`, and the general way to do this kind of experiment

### CODE REVIEW

File: components/dialogs/AuthDialog.vue [hunk 1]
```diff
@@ -32,7 +32,16 @@
...
-      <img :src="LikeCoinLogo">
+      <img
+        v-if="avatar"
+        :src="avatar"
+        class="auth-dialog__avatar auth-dialog__avatar--user"
+      >
+      <img
+        v-else
+        :src="LikeCoinLogo"
+        class="auth-dialog__avatar auth-dialog__avatar--likecoin"
```
Comment (nwingt) [inline components/dialogs/AuthDialog.vue]: I prefer to name this class as `auth-dialog__logo`, with modifier `--avatar` and `--likecoin`.

I suggest to add a computed property i.e. `logoProps` which return
```javascript
{
  src: this.shouldShowFromUserIcon ? this.avatar : LikeCoinLogo,
  class: `auth-dialog__logo auth-dialog__logo--${this.shouldShowFromUserIcon ? 'avatar' : 'likecoin'}`,
}
```
so we can use `v-bind="logoProps"` with one ``

File: components/dialogs/AuthDialog.vue [hunk 2]
```diff
@@ -291,6 +302,12 @@ export default {
         this.$t('AuthDialog.label.signInError')
       );
     },
+    shouldShowFromUserIcon() {
+      if (!this.$exp) return false;
+      const { name, $activeVariants } = this.$exp;
+      return name === 'register-icon'
+        && !!$activeVariants.find(variant => variant.name === 'from-user-avatar');
+    },
```
Comment (nwingt) [inline components/dialogs/AuthDialog.vue]: I suggest to create a functional mixin for later use

e.g. 
```javascript
function abTestMixin(
  name,
  expName,
  varName,
) {
  return {
    computed: {
      [name]() {
        if (!this.$exp) return false;
        const { name, $activeVariants } = this.$exp;
        return name === expName
          && !!$activeVariants.find(variant => variant.name === varName);
      },
    },
  };
}

// AuthDialog.vue
...
mixins: [
  abTestMixin({
    shouldShowFromUserIcon,
    'register-icon',
    'from-user-avatar'
  })
],
...
```

File: components/dialogs/AuthDialog.vue [hunk 3]
```diff
@@ -325,6 +342,18 @@ export default {
     },
   },
   async mounted() {
+    if (this.shouldShowFromUserIcon) {
```
Comment (nwingt) [inline components/dialogs/AuthDialog.vue]: Is `shouldShowFromUserIcon` ready when executing `mounted()`?

Comment (williamchong007) [reply components/dialogs/AuthDialog.vue]: `shouldShowFromUserIcon` is populated by nuxt plugin, thus should be always ready

File: components/dialogs/AuthDialog.vue [hunk 4]
```diff
@@ -291,6 +301,12 @@ export default {
         this.$t('AuthDialog.label.signInError')
       );
     },
+    logoProps() {
+      return {
+        src: this.shouldShowFromUserIcon ? this.avatar : LikeCoinLogo,
```
Comment (nwingt) [inline components/dialogs/AuthDialog.vue]: if avatar default is LikeCoinLogo, can simply return `src: this.avatar`?

**Pull request merged**